src/app/validators/nation-validator.service.ts

```diff
@@ -15,9 +15,9 @@ export class NationValidatorService {
   readonly dbName = 'nations';
 
   constructor(private couchService: CouchService) {}
-    public nationCheckerService$(name: string): Observable<boolean> {
+    public nationCheckerService$(name: string, nationUrl: string): Observable<boolean> {
       const isDuplicate = this.couchService
-      .post(`${this.dbName}/_find`, findOneDocument('name', name))
+      .post(`${this.dbName}/_find`, findAllDocuments('name', name, 'nationUrl', nationUrl))
```

Comment:
This will not work, since `findAllDocuments` only takes two arguments: one `selector` and one `query`.